### DISCUSSION TITLE
Developer GraphQL returns "Get query missing"

### DISCUSSION THREAD

**Discussion by kurtis318** [requarks/wiki#3254] [Error / Bug Report]
### Actual behavior
GET query missing.
### Expected behavior
Should see GraphQL playground 
### Steps to reproduce the behavior
Sign in as Admin
Click on Administration cog icon
Expand Developer Tools in left navigator
Click on GraphQL
See this page:
![image](https://user-images.githubusercontent.com/8187747/99753203-6f4ca400-2aab-11eb-8183-250e0aeb69ef.png)

**Comment by ghost** [comment]
You haven't provided the required info about your host! (OS, Wiki.js version, Database engine)

**Comment by kurtis318** [comment]
Sorry,

OS=Linux - Fedora (Thirty Three) 33 x64
Wiki.js version=2.5.170
Database engine=PostgreSQL 12.4
Node.js version=14.14.0

**Comment by NGPixel** [comment]
Are you loading `/graphql`?

**Comment by kurtis318** [comment]
Yes, I am loading /graphql from the Wiki.js interface I described above.  The URL is hostname:3000/graphql .
This was working on Fedora 32 Sever but Fedora 33 probably has a newer Node.js version.

**Comment by liziker** [comment]
I have the same issue...

OS=Linux - Ubuntu Server (focal) 20.04 x64
Wiki.js version=2.5.170
Database engine=(PostgreSQL) 13.1 (Ubuntu 13.1-1.pgdg20.04+1）
Node.js version=14.15.4

**Comment by liziker** [comment]
I have the same issue...

OS=Linux - Ubuntu Server (focal) 20.04 x64
Wiki.js version=2.5.170
Database engine=(PostgreSQL) 13.1 (Ubuntu 13.1-1.pgdg20.04+1）
Node.js version=14.15.4

**Comment by jcv99** [comment]
Same
OS= debian 10
wiki.js = 2.5.170
database= mysql 8.0
node js= 10.24.0

i modiefied a file in graphql, but i remodified. And I wrote somethig in the editor of the graph before i couldn't enter,something like :
somthing{
<TAB>maxfilesize:50000000
}
i dont know if that broke it (i dont think but...)

**Comment by exglade** [comment]
Wiki.js uses Apollo, according to the [documentation](https://www.apollographql.com/docs/apollo-server/testing/graphql-playground/), if your environment is production, then you won't get GraphQL playground.

I hope that explains the behaviour.

~~On the other hand, I don't mind not having playground, but I couldn't even get it working when sending GraphQL query with Postman. I get the same error message as everyone:~~

> GET query missing.

~~I made sure my HTTP method is POST. 🤷‍♀️~~

Edit: My POST request was redirected by HTTP 301, hence turning into GET. My mistake, there is no issue for me.

**Comment by jantari** [comment]
I figured out a way to enable the playground - however this is not ideal as it will get overwritten when wiki.js updates.
Since the wiki.js documentation points to the playground as the only API documentation, but they playground doesn't actually work, I think this is an issue and it needs to be fixed (playground enabled by default or provide API documentation on the docs website) - either way, do this:

in the file: `$WIKIROOT/server/core/servers.js` find the function:

```js
/**
* Start GraphQL Server
*/
async startGraphQL () {
    // truncated //
```

for me this was around line 120 in the file. Inside this function find the call to `new ApolloServer`. For me it looked like this:

```js
this.servers.graph = new ApolloServer({
    // truncated list of options //
})
```

as one of the options inside this `new ApolloServer({})` call, add:

```js
playground: true,
introspection: true,
```

Make sure you have the trailing comma unless you add it as the last option.

Then restart the wikijs service/process - the playground now works and appears when accessing `/graphql` in a browser.

---

In case this ever becomes outdated in a future wiki.js version you can always search for any calls to `new ApolloServer` with GNU grep - that's how I found this too:

```bash
cd "$WIKIROOT"
grep -ri "new ApolloServer" .
```

**Reply by kurtis318** [reply]
Just updated my server and now it works.  Thanks for sharing.

**Comment by ghost** [comment]
Can confirm I'm also having this issue. Accessing wiki.mydomain.com/graphql returns "GET query missing".

Wiki.js: 2.5.201
CentOS 8.4.2015 x64
NodeJS 14.16.0
PostgreSQL 10.17

**Comment by rrg92** [comment]
i'm having this issue also.

Wiki.js: 2.5.286
Ubuntu 20.04
node v16.15.0
PostgreSQL 12.12

**Reply by rrg92** [reply]
I found the problem.
I'm running my server using NODE_ENV=production. 
Apollo checks this environment var to decide if will enable playground or not buy default.
